src/hotspot/share/runtime/os.cpp

```diff
@@ -1487,7 +1487,9 @@ static const char* errno_to_string (int e, bool short_text) {
     X(ENETUNREACH, "Network unreachable") \
     X(ENFILE, "Too many files open in system") \
     X(ENOBUFS, "No buffer space available") \
+#ifdef ENODATA
```

Comment:
I really wonder how this is working on the current platforms, since x86 FreeBSD does not have ENODATA defined either.

Anyway, the correct solution is to move these entries into individual added sections like those starting at line 1537 in the current version of the file.  It looks like this:

```
    // The following enums are not defined on all platforms.
    #ifdef ESTALE
    DEFINE_ENTRY(ESTALE, "Reserved")
    #endif
```

etc.